### ISSUE TITLE
Controller over reflex attribute is ignore and reflex is always called with default StimulusReflexController

### ISSUE THREAD

**Issue by Laykou** [stimulusreflex/stimulus_reflex#659]
# Bug Report

## Describe the bug

Using the custom `data-controller` above `data-reflex` is ignored and reflex is always executed with default `stimulus-reflex` controller.

This breaks behavior and callbacks described in https://docs.stimulusreflex.com/guide/lifecycle.html#callback-methods

It may be related to: https://github.com/stimulusreflex/stimulus_reflex/pull/636

Discussed in Discord: https://discord.com/channels/629472241427415060/733725826411135107/1100724714760044606

## To Reproduce

```html
<div data-controller="example">
  <a href="#" data-reflex="Example#masticate">Eat</a>
</div>
```

```js
import ApplicationController from './application_controller.js'

export default class extends ApplicationController {
  beforeReflex(element) {
    console.log('Called here');
  }
}
```

In the end the element has `<a href="#" data-reflex="Example#masticate" data-action="stimulus-reflex#__perform">Eat</a>`  tag. Console log is not shown

## Expected behavior

Element should have `<a href="#" data-reflex="Example#masticate" data-action="example#__perform">Eat</a>` tags

## Screenshots or reproduction

- pre10 - where it worked
  ![image](https://user-images.githubusercontent.com/2313005/234763525-b0ca16bf-cefe-4d03-b94a-8e4c527e801a.png)

- rc1 - where it's not working
  ![image](https://user-images.githubusercontent.com/2313005/234763547-9fe13dd9-965c-4992-8b4b-040cd8f3e6f7.png)

## Versions

- pre9 - worked
- pre10 - worked
- rc1 - not working
- rc2 - not working

**Comment by Laykou** [comment]
In our case the hotfix is creating a method in JS `data-action="my-controller#myMethod"` and call `this.stimulate('Reflex#action', event.target)` within it manually.

**Comment by Laykou** [comment]
@julianrubisch Is this still an open topic or was this issue somehow addressed/fixed in the latest changes in the past month?

I'd like to know if it's worth upgrading the lib or if we still have to use workaround via custom actions `data-action` + `this.stimulate`

**Comment by marcoroth** [comment]
I need to double check, it might be related #636, but I think it's more related to an unhandled edge case with setting up the declarative reflex attributes.

**Comment by Laykou** [comment]
Is there some new version/release I could try this on?

**Comment by marcoroth** [comment]
Looks like I can't reproduce this on rc2. 

But looking at your example you need to make sure also include the `event` itself in your `data-reflex` attribute:
```diff
- data-reflex="Example#masticate"
+ data-reflex="click->Example#masticate"
```

So the following snippet:
```js
<div data-controller="example">
  <a href="#" data-reflex="click->Example#masticate">Eat</a>
</div>
```
will transform into this as expected (this is on `3.5.0.rc2`):
```html
<div data-controller="example">
  <a href="#" data-reflex="click->Example#masticate" data-action="click->example#__perform">Eat</a>
</div>
```

**Comment by marcoroth** [comment]
This should be resolved with the release of [`v3.5.0.rc4`](https://github.com/stimulusreflex/stimulus_reflex/releases/tag/v3.5.0.rc4). I'm going to close this issue, but please feel free to re-open if you are still seeing something unexpected!
